internal/builders/maven/plugins/verification-plugin/src/main/java/io/github/adamkorcz/SlsaVerificationMojo.java

```diff
@@ -0,0 +1,189 @@
+                executeMojo(
+                    plugin(
+                        groupId("com.googlecode.maven-download-plugin"),
+                        artifactId("download-maven-plugin"),
+                        version("1.7.0")
+                    ),
+                    goal("artifact"),
+                    configuration(
+                        element(name("outputDirectory"), "${project.build.directory}/slsa"),
+                        element(name("groupId"), artifact.getGroupId()),
+                        element(name("artifactId"), artifact.getArtifactId()),
+                        element(name("version"), artifact.getVersion()),
+                        element(name("type"), "intoto.build.slsa"),
+                        element(name("classifier"), "jar")
+                    ),
+                    executionEnvironment(
+                        project,
+                        mavenSession,
+                        pluginManager
+                    )
+                );
+
+                // Retrieve the dependency jar if slsa file does exists for this artifact
+                executeMojo(
+                    plugin(
+                        groupId("org.apache.maven.plugins"),
+                        artifactId("maven-dependency-plugin"),
+                        version("3.6.0")
+                    ),
+                    goal("copy"),
+                    configuration(
+                        element(name("outputDirectory"), "${project.build.directory}/slsa"),
+                        element(name("artifact"), artifactStr)
+                    ),
+                    executionEnvironment(
+                        project,
+                        mavenSession,
+                        pluginManager
+                    )
+                );
+            } catch(MojoExecutionException e) {
+                getLog().info("Skipping slsa verification for " + artifactStr + ": No slsa file found.");
+                continue;
+            }
+
+            // Verify slsa file
+            try {
+                // Run slsa verification on the artifact and print the result
+                // It will never fail the build process
+                String arguments = "verify-artifact --provenance-path ";
+                arguments += "${project.build.directory}/slsa/" + artifact.getArtifactId() + "-" + artifact.getVersion() + "-jar.intoto.build.slsa ";
+                arguments += " --source-uri ./ ${project.build.directory}/slsa/" + artifact.getArtifactId() + "-" + artifact.getVersion() + ".jar";
+                executeMojo(
+                    plugin(
+                        groupId("org.codehaus.mojo"),
+                        artifactId("exec-maven-plugin"),
+                        version("3.1.0")
+                    ),
+                    goal("exec"),
+                    configuration(
+                        element(name("executable"), gohome + "/bin/slsa-verifier"),
```

Comment:
if we let the user installs, is there a consistent way to get the path to the slsa-verifier?

Reply:
The user now passes a parameter when using the plugin. This sets an environment variable that contains the path to the slsa-verifier.

Reply:
if the binari is in the `$PATH`, the user need no provide anything, correct? For xeample, if a user install using the GitHub action in a workflow, they won't know where it was installed. They just know it's in the path and they can run `$ slsa-verifier`. In Golang, you can use https://pkg.go.dev/os/exec#LookPath to get the full path to a binary. Is there something similar for Java?

Reply:
gotcha... let's discuss in follow-up PR then.